app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -68,10 +69,23 @@ class MainActivity : AppCompatActivity() {
             }
             R.id.orderCompletedFragment -> navController.popBackStack(R.id.eventDetailsFragment, false)
             R.id.welcomeFragment -> finish()
+            R.id.editProfileFragment -> {
+
+                // Calls the handleBackPress method in EditProfileFragment
+                val hostFragment = supportFragmentManager.findFragmentById(R.id.frameContainer) as? NavHostFragment
+                (hostFragment?.childFragmentManager?.fragments?.get(0) as? EditProfileFragment)?.handleBackPress()
+            }
             else -> super.onBackPressed()
         }
     }
 
+    /**
+     * Called by EditProfileFragment to go to previous fragment
+     */
+    fun onSuperBackPressed() {
+        super.onBackPressed()
```

Comment:
Useless as it's just calling onBackPressed. OnBackPressed can be called directly

Reply:
I use this method to call the direct onBackPressed from the EditProfileFragment. If i would have called the onBackPressed then it would have called the overriden method which would again display the AlertDialog

Reply:
Ok

Reply:
Any changes I should make?